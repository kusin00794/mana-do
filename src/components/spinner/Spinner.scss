@import '../../styles/_zIndex.scss';
@import '../../styles/variables.scss';

.spinner-loading-wrapper {
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  width: 100%;
  height: 100%;
  align-items: center;
  justify-content: center;
  background: $white-color;
  z-index: $zIndexSpinnerLoading;
}

.loader {
  position: relative;
  width: 15em;
  height: 15em;
  margin: 50px auto;
  border-radius: 50%;
  font-size: 5px;
  text-indent: -9999em;
  transform: translateZ(0);
  animation: spinner 1.4s infinite linear;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    content: '';
    width: 50%;
    height: 50%;
    border-radius: 100% 0 0 0;
    background: $spinner-color;
  }

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    content: '';
    width: 75%;
    height: 75%;
    margin: auto;
    border-radius: 50%;
    background: $white-color;
  }
}

@-webkit-keyframes spinner {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes spinner {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
