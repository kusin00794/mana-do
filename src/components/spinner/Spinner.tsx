import React, { FC } from 'react';

import './Spinner.scss';

const Spinner: FC = () => {
  return (
    <div className="spinner-loading-wrapper">
      <div className="loader">Loading...</div>
    </div>
  );
};

export default Spinner;
