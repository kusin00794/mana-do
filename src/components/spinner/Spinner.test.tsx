import { testSnapshots } from '../../utils/test';
import Spinner from './Spinner';

describe('<Spinner />', () => {
  testSnapshots(Spinner, [
    {
      props: {},
      description: 'default render',
    },
  ]);
});
