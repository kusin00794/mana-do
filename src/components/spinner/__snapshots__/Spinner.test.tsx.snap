// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> snapshots default render 1`] = `
<div
  className="spinner-loading-wrapper"
>
  <div
    className="loader"
  >
    Loading...
  </div>
</div>
`;
